### DISCUSSION TITLE
Kotlin compose inspired UI

### DISCUSSION THREAD

**Discussion by ronjunevaldoz** [libktx/ktx#439] [Ideas]
For me, creating scene2d GUI is a bit complex, so I came to an idea to have a compose UI inspired DSL maybe using `ktx-scene2d`  and `scene2d` as well.

![image info](https://developer.android.com/static/images/jetpack/compose/landing-code-static.svg)

https://developer.android.com/jetpack/compose

**Comment by czyzby** [comment]
`ktx-scene2d` and `ktx-vis` already wrap Scene2D and VisUI widgets with a Kotlin DSL that looks somewhat similar to Compose. I'm not sure what else we could do outside of renaming the DSL functions to match the Compose equivalents. Other features such as state handling with automatic UI updates, animated transitions etc. would have to be reimplemented from scratch, since Scene2D does not support those. Do you propose adding anything specific that would make it easier for you to use the existing GUI modules?
